backend/data/utility.py

```diff
@@ -21,3 +21,52 @@ def create_delta_cols( df,group_by_cols, delta_needed_for_cols):
     df = df.set_index(index_cols)
 
     return df
+
+
+def interpolate_values(df, group_by_cols, delta_needed_for_cols):
```

Comment:
add some python type hints and short doc string as what this function does.

Reply:
added short doc string